src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -1,16 +1,103 @@
 package thurloe.database
 
-import thurloe.service.KeyValuePair
+import com.typesafe.config.ConfigFactory
+import scala.concurrent.{Future, Await}
+import scala.concurrent.ExecutionContext.Implicits.global
+import thurloe.service.{UserKeyValuePairs, UserKeyValuePair, KeyValuePair}
 
-import scala.util.{Success, Try}
+import scala.concurrent.duration.Duration
+import scala.util.{Failure, Success, Try}
 
 case object ThurloeDatabaseConnector extends DataAccess {
 
-  // TODO: Fill THESE in with database access implementation.
-  def keyLookup(key: String) = Success(KeyValuePair("yek", "eulav"))
-  def collectAll() = Success(Seq(
-    KeyValuePair("key1", "Bob Loblaw's Law Blog"),
-    KeyValuePair("key2", "Blah blah blah blah blah")))
-  def setKeyValuePair(keyValuePair: KeyValuePair): Try[Unit] = Success(())
-  def deleteKeyValuePair(key: String): Try[Unit] = Success()
+  val config = ConfigFactory.load().getConfig("database")
+  val databaseConfig = config.getConfig(config.getString("config"))
+
+  val dataModels = new DatabaseDataModels(databaseConfig.getString("slick.driver"))
+
+  import dataModels._
+  import dataModels.driver.api._
+
+  val database = Database.forConfig("", databaseConfig)
+
+  if (databaseConfig.hasPath("slick.createSchema") && databaseConfig.getBoolean("slick.createSchema"))
+    setupInMemoryDatabase(database)
+
+  private def lookupWithConstraint(constraint: DbKeyValuePair => Rep[Boolean]): Future[Seq[KeyValuePair]] = {
+    val keyValuePairs = TableQuery[DbKeyValuePair]
+    val q = keyValuePairs.filter(constraint)
```

Comment:
could probably use a better name than just `q` unless that's some kind of convention I'm not aware of